modules/tinymce/src/core/main/ts/dom/TrimNode.ts

```diff
@@ -5,26 +5,26 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Arr } from '@ephox/katamari';
+import { Type } from '@ephox/katamari';
 import { SugarElement } from '@ephox/sugar';
 import DOMUtils from '../api/dom/DOMUtils';
-import { isWhitespaceText } from '../text/Whitespace';
+import DomTreeWalker from '../api/dom/TreeWalker';
 import * as ElementType from './ElementType';
+import { isContent } from './Empty';
 import * as NodeType from './NodeType';
-import * as Empty from './Empty';
 
 const isSpan = (node: Node): node is HTMLSpanElement =>
   node.nodeName.toLowerCase() === 'span';
 
-const isNonEmptyText = (node: Node) =>
-  NodeType.isText(node) && !isWhitespaceText(node.data);
+const isInlineContent = (node: Node | null, root: Node) =>
+  node && (isContent(root, node) || ElementType.isInline(SugarElement.fromDom(node)));
 
-const isInlineContent = (node: Node | null) =>
-  node && (isNonEmptyText(node) || ElementType.isInline(SugarElement.fromDom(node)));
-
-const surroundedByInlineContent = (node: Node) => {
-  const prevIsInline = isInlineContent(node.previousSibling);
-  const nextIsInline = isInlineContent(node.nextSibling);
+const surroundedByInlineContent = (node: Node, root: Node) => {
+  const prev = new DomTreeWalker(node, root).prev(false);
+  const next = new DomTreeWalker(node, root).next(false);
+  // Check if the next/previous is either inline content or the start/end (eg is undefined)
+  const prevIsInline = isInlineContent(prev, root) || Type.isUndefined(prev);
```

Comment:
Could we take advantage of the short-circuiting and put the "is it falsey" check first so that the `isInlineContent` function doesn't need to take a `Node | null` maybe? Or (and this may be overkill) go one step further and wrap the prev / next definitions with `Optional.from`.

Worth noting that using `&&` everywhere for falsey checks means that these functions technically return `boolean | null` rather than just `boolean` at the moment.

Reply:
`Optional.from` !== `Type.isUndefined`, so no you can't do that as they have different functionality (I could have used `Type.isNullable` otherwise). With regards to the original point, it'd be a micro optimisation that shouldn't hold much benefit, but I might as well make the change.

Fair point about the return type being `boolean | null`. That was like that previously and is like that in quite a number of places tbh, but I'll fix this specific case up.